src/test/regress/input/appendonly.source

```diff
@@ -845,3 +845,34 @@ explain select cmax, * from test_reporting_r;
 drop table test_reporting;
 drop table test_reporting_r;
 drop table test_reporting_c;
+
+-- Test AO functions with execution location on dispatcher.
+-- start_ignore
+drop table if exists t1;
+drop table if exists t2;
+drop table if exists tables_list;
+-- end_ignore
+
+create table t1 (a int) with (appendoptimized=true, orientation=row);
+create table t2 (a int) with (appendoptimized=true, orientation=column);
+create table tables_list (tname text);
+insert into t1 values (0);
+insert into t2 values (0);
```

Comment:
Adding rows to `t1` and `t2` is extra. Error messages are printed without it.

Reply:
> Adding rows to t1 and t2 is extra. Error messages are printed without it.

Ok.

But it uncovered another issue. 
For an empty column-oriented table `get_ao_compression_ratio` returns -1. I consider that an expected results, as according to function specification, it:
```
Calculates the compression ratio for a compressed append-optimized table. 
If information is not available, this function returns a value of -1.
```
But for an empty row-oriented table `get_ao_compression_ratio` returns 1. It comes from the code in `aorow_compression_ratio_internal`, which is some fix for an old compatibility issue with GPDB 3.3 tables:
```
			...
			/*
			 * Get totals from QE's and calculate the compression ratio. In
			 * tables upgraded from GPDB 3.3 the eofuncompressed column could
			 * contain NULL, this is fixed in more recent upgrades.
			 */
			char	   *attr1 = SPI_getvalue(tuple, tupdesc, 1);
			char	   *attr2 = SPI_getvalue(tuple, tupdesc, 2);

			if (NULL == attr1 || NULL == attr2)
			{
				compress_ratio = 1;
			}
			else if (scanint8(attr1, true, &eof) &&
					 scanint8(attr2, true, &eof_uncomp))
			...
```

I think that for an empty table the output of `get_ao_compression_ratio` should be consistent for both column- and row-oriented tables.
So I've updated this part of function as well.

Reply:
Decided to discard the changes in `get_ao_compression_ratio` for empty row-oriented tables as they are not directly related to the current issue and they affect another test.